### DISCUSSION TITLE
Q: What exactly does `LAPACKE_WITH_TMG=ON`?

### DISCUSSION THREAD

**Discussion by kloczek** [Reference-LAPACK/lapack#716] [Q&A]
Somoene know what exactly does `LAPACKE_WITH_TMG=ON`? 🤔

**Comment by langou** [comment]
I believe TMG stands for "Test Matrix Generation". When you build LAPACK, you have the option to create libtmg. libtmg is a library of subroutines that generates matrices that are useful for testing purpose. The folder is TESTING/MATGEN. There are various weird matrices. Various structures (general, banded, etc.), various properties (general, SPD, singular, etc.), various scaling (near overflow, near underflow, etc.), some difficult pencil for generalized eigenvalue problems, etc. and then some classic matrices such as the Hilbert matrix. 

This library is useful and enables breaking the test suite in three parts. (1) On the one hand 🤚 , the creation of matrices. (2) On the other hand 🤚, the creation of tester for a given LAPACK functionality. (3) And on the third hand 🤚 (?? I messed up here 😂 ??), a driver that runs all the testers on all the matrices. Kind of the idea. 

I am assuming that ``LAPACKE_WITH_TMG=ON`` means to ask to create the libtmg library.

**Comment by kloczek** [comment]
Thankl you very much 👍 
I think that it would be good to drop such srort description somewhere in README 😋 

So TNG should be enabled when `BUILD_TESTING=ON`?
If yes probably use `BUILD_TESTING=ON` should automatically trigger `LAPACKE_WITH_TMG=ON` (?)

Is it anything outside lapack which is using that TNG library?
I'm asking because I see that it is installed when `LAPACKE_WITH_TMG=ON`.
